In [28]:
import csv
f = open("guns.csv", "r")
csvreader = csv.reader(f)
data = list(csvreader)
headers = data[:1]
data = data[1:]
print(headers)

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']]


In [29]:
print(data[:5])

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [30]:
years = [row[1] for row in data]
year_counts = {}
for year in years:
    if year in year_counts:
        year_counts[year] += 1
    else:
        year_counts[year] = 1
print(year_counts)

{'2014': 33599, '2013': 33636, '2012': 33563}


In [31]:
import datetime
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data]
print(dates[:5])

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


In [32]:
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1

highest_deaths = 0
lowest_deaths = 5000
for date, value in date_counts.items():
    if int(value) > highest_deaths:
        highest_deaths = value
        highest_date = date
    elif int(value) < lowest_deaths:
        lowest_deaths = value
        lowest_date = date
print(highest_deaths)
print(highest_date)

print(lowest_deaths)
print(lowest_date)

3079
2013-07-01 00:00:00
2357
2012-02-01 00:00:00


In [89]:
def column_counts(column):
    column_count_dict = {}
    for row in data:
        if row[column] in column_count_dict:
            column_count_dict[row[column]] += 1
        else:
            column_count_dict[row[column]] = 1
    return column_count_dict
        
print(column_counts(5))
print(column_counts(3))
print(column_counts(9))
print(column_counts(10))

{'M': 86349, 'F': 14449}
{'Undetermined': 807, 'Homicide': 35176, 'Suicide': 63175, 'Accidental': 1639, 'NA': 1}
{'Farm': 470, 'NA': 1384, 'Other unspecified': 8867, 'Industrial/construction': 248, 'Trade/service area': 3439, 'Street': 11151, 'School/instiution': 671, 'Home': 60486, 'Other specified': 13751, 'Sports': 128, 'Residential institution': 203}
{'NA': 53, '1': 21823, '3': 21680, '2': 42927, '5': 1369, '4': 12946}


In [34]:
race_counts = {}
for row in data:
    if row[7] in race_counts:
        race_counts[row[7]] += 1
    else:
        race_counts[row[7]] = 1

print(race_counts)

{'Native American/Native Alaskan': 917, 'Black': 23296, 'White': 66237, 'Asian/Pacific Islander': 1326, 'Hispanic': 9022}


# Interesting Observations

**Males much more likely to die in gun death**

**Blacks and Hispanics seem more likely to die compared to population percentages, but need population info to verify**

** Deaths seem to be generally higher in the summer than the winter**

In [35]:
file2 = open("census.csv", "r")
file2_reader = csv.reader(file2)
census = list(file2_reader)
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [53]:
mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Black": 40250635,
    "Native American/Native Alaskan": 3739506,
    "Hispanic": 44618105,
    "White": 197318956
}

race_per_hundredk = {}
for each in race_counts:
    race_per_hundredk[each] = float(race_counts[each]) / float(mapping[each]) * 100000
print(race_per_hundredk)

{'Native American/Native Alaskan': 24.521955573811088, 'Asian/Pacific Islander': 8.374309664161762, 'White': 33.56849303419181, 'Black': 57.8773477735196, 'Hispanic': 20.220491210910907}


In [56]:
intents = [row[3] for row in data]
races = [row[7] for row in data]

homicide_race_counts = {}
for i, race in enumerate(races):
    if intents[i] == "Homicide":
        if races[i] in homicide_race_counts:
            homicide_race_counts[races[i]] += 1
        else:
            homicide_race_counts[races[i]] = 1
        
print(homicide_race_counts)

{'Native American/Native Alaskan': 326, 'Black': 19510, 'White': 9147, 'Asian/Pacific Islander': 559, 'Hispanic': 5634}


In [57]:
for each in homicide_race_counts:
    homicide_race_counts[each] = float(homicide_race_counts[each]) / float(mapping[each]) * 100000

print(homicide_race_counts)

{'Native American/Native Alaskan': 8.717729026240365, 'Black': 48.471284987180944, 'White': 4.6356417981453335, 'Asian/Pacific Islander': 3.530346230970155, 'Hispanic': 12.627161104219914}


## Interesting Notes

*Quite a few more Homicide deaths in the Black race per 100K populants than any other race.*

**Homicide Deaths per 100K populants:**
- Black: 48.5
- Hispanic: 12.6
- Native American/Alaskan: 8.7
- White: 4.6
- Asian/Pacific Islander: 3.5

**For follow up: **
- Need to investigate Police correlation with race
- Need to investigate place and intent correlation

**Suggested investigation: **
- Figure out the link, if any, between month and homicide rate.
- Explore the homicide rate by gender.
- Explore the rates of other intents, like Accidental, by gender and race.
- Find out if gun death rates correlate to location and education.

In [39]:
police_involved_data = [row for row in data if int(row[4]) == 1]

print(len(police_involved_data))

1402


### Not as much police to race correlation as I would have thought

In [62]:
police_race_counts = {}
for row in police_involved_data:
    if row[7] in police_race_counts:
        police_race_counts[row[7]] += 1
    else:
        police_race_counts[row[7]] = 1

print(police_race_counts)

police_race_counts_per100k = {}
for each in police_race_counts:
    police_race_counts_per100k[each] = float(police_race_counts[each]) / float(mapping[each]) * 10000000

print(police_race_counts_per100k)

{'Native American/Native Alaskan': 25, 'Hispanic': 282, 'White': 709, 'Asian/Pacific Islander': 30, 'Black': 356}
{'Native American/Native Alaskan': 66.85375020122979, 'Hispanic': 63.20304280067475, 'White': 35.93167196769478, 'Asian/Pacific Islander': 18.94640195511711, 'Black': 88.44580961269307}


### Let's see about Homicide rate by gender
It would appear that males are much more involved in gun deaths across all intents. 
- Suicide and Accidental are slightly more skewed to males than Homicide

In [83]:
intents = [row[3] for row in data]
genders = [row[5] for row in data]

def intent_by_gender(intent, intents, genders):
    intent_by_gender_counts = {"Intent": intent}
    for i, gender in enumerate(genders):
        if intents[i] == intent:
            if genders[i] in intent_by_gender_counts:
                intent_by_gender_counts[genders[i]] += 1
            else:
                intent_by_gender_counts[genders[i]] = 1
    intent_by_gender_counts["ratio"] = float(intent_by_gender_counts["M"])/float(intent_by_gender_counts["F"])
    return intent_by_gender_counts
        
homicide_by_gender = intent_by_gender("Homicide", intents, genders)
print(homicide_by_gender)
print(intent_by_gender("Accidental", intents, genders))
print(intent_by_gender("Suicide", intents, genders))
print(intent_by_gender("Undetermined", intents, genders))

{'M': 29803, 'Intent': 'Homicide', 'ratio': 5.546808114647311, 'F': 5373}
{'M': 1421, 'Intent': 'Accidental', 'ratio': 6.518348623853211, 'F': 218}
{'M': 54486, 'Intent': 'Suicide', 'ratio': 6.270687075612844, 'F': 8689}
{'M': 638, 'Intent': 'Undetermined', 'ratio': 3.775147928994083, 'F': 169}
